apps/condo/domains/user/schema/RegisterNewUserService.js

```diff
@@ -43,51 +43,43 @@ const RegisterNewUserService = new GQLCustomSchema('RegisterNewUserService', {
-
-                    await ensureNotExists(context, 'User', 'Users', 'phone', phone_number)
-                    await ensureNotExists(context, 'User', 'Users', 'importId', uid)
-
-                    userData.phone = phone_number
-                    userData.isPhoneVerified = true
-                    userData.importId = uid
-                }
-
-                // TODO(Dimitreee): use ensureNotExists
-                const { errors: findErrors, data: findData } = await context.executeGraphQL({
-                    context: context.createContext({ skipAccessControl: true }),
-                    query: `
-                            query findUserByEmail($email: String!) {
-                              users: allUsers(where: { email: $email }) {
-                                id
-                              }
-                            }
-                        `,
-                    variables: { email: userData.email },
-                })
-
-                // TODO(Dimitreee): add usage of guards
-
-                if (findErrors) {
-                    const msg = '[error] Unable to call find service'
-                    throw new Error(msg)
-                }
-
-                if (findData.users.length !== 0) {
-                    throw new Error(`${EMAIL_ALREADY_REGISTERED_ERROR}] User with this email is already registered`)
+                if (confirmPhoneActionToken) {
+                    const { errors: findConfirmPhoneActionErrors, data: confirmActions } = await context.executeGraphQL({
+                        context: context.createContext({ skipAccessControl: true }),
+                        query: `
+                                query findConfirmPhoneActionsByToken($token: String!) {
+                                  actions: allConfirmPhoneActions(where: { token: $token }) {
+                                    phone
+                                    isPhoneVerified
+                                  }
+                                }
+                            `,
+                        variables: { token: confirmPhoneActionToken },
+                    })
+                    if (findConfirmPhoneActionErrors || !confirmActions || isEmpty(confirmActions.actions)) {
+                        const msg = '[error] Unable to find confirm phone action'
+                        throw new Error(msg)    
+                    }
+                    const { phone, isPhoneVerified } = confirmActions.actions[0]
+                    if (!isPhoneVerified) {
+                        const msg = '[error] Phone is not verified'
+                        throw new Error(msg)    
+                    }
+                    userData.phone = phone
+                    userData.isPhoneVerified = isPhoneVerified
                 }
+                // TODO(zuch): Ask if we allow users without phone or/and email to register
+                await ensureNotExists(context, 'User', 'Users', 'phone', userData.phone)
```

Comment:
Email/phone required. We couldn't allow users to register during they'll provide valid and correct data. Sadly we cannot validate email at the start of registration now:(

Reply:
Added check for value presence in ensureNotExists